test/extensions/filters/http/fault/fault_filter_test.cc

```diff
@@ -580,7 +580,11 @@ TEST_F(FaultFilterTest, FixedDelayDeprecatedPercentAndNonZeroDuration) {
   EXPECT_EQ(1UL, config_->stats().active_faults_.value());
   EXPECT_EQ(Http::FilterTrailersStatus::StopIteration, filter_->decodeTrailers(request_trailers_));
   EXPECT_CALL(decoder_filter_callbacks_, continueDecoding());
+
   timer_->invokeCallback();
+
+  EXPECT_EQ(0UL, config_->stats().active_faults_.value());
```

Comment:
The behavior change here is not just for stats; it's also about the max number of consecutive faults allowed.  I suggest adding a test that sets the max number of faults to 1 and delays 100% of requests, starts a streaming call, and after the streaming call has been delayed, starts a second call.  That second call should also be delayed (without this fix, it would not be).

Reply:
Thanks for the comments! I do agree.
One question - should we add this kind of test to sandbox level or were you suggesting it for unit level?

Reply:
I don't have a strong feeling about what kind of test is best here.  I'll let one of the Envoy maintainers answer that.

Reply:
@asraa Would you please kindly help give some advice here so that we could progress this PR? Thanks.

Reply:
I think a unit test that you can
start request 1 with a fault delay
finish request 1 delay (but not request 1)
start request 2 and have the fault delay kick in
would be fine.  As mark said you'll want to make sure you set the limit on active faults - generally when writing a regression test like this I double check to make sure it's testing the right thing by making sure it fails without my changes :-)

Reply:
Thanks for the hints and suggestions! Updated with a new attempt.

Reply:
Looks great!  Do you mind adding a regression test that would catch issues like this in future?
I think you can mostly crib off your new test, only have a response limiter active too, and make sure it doesn't free up.

Reply:
I'd love to add.
However, it seems to me that making a second request w/o ending the first one would be a little bit hard to do in integration tests (e.g., for new streams, the codec will make sure there's no pending response).
Would you please kindly help point me to some reference if there's any?

Reply:
> Looks great! Do you mind adding a regression test that would catch issues like this in future?
> I think you can mostly crib off your new test, only have a response limiter active too, and make sure it doesn't free up.

Let me know if the new test added matches the intention of the suggestion here. Thanks.